jaeger-dropwizard/src/main/java/com/uber/jaeger/dropwizard/JerseyServerFilter.java

```diff
@@ -0,0 +1,117 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.uber.jaeger.dropwizard;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.uber.jaeger.context.TraceContext;
+import com.uber.jaeger.filters.jaxrs2.ServerFilter;
+import io.opentracing.Tracer;
+import org.glassfish.jersey.server.ContainerRequest;
+import org.glassfish.jersey.server.ExtendedUriInfo;
+import org.glassfish.jersey.server.internal.routing.UriRoutingContext;
+import org.glassfish.jersey.uri.UriTemplate;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.ws.rs.Path;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.ext.Provider;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A {@link ServerFilter} that provides a Jersey specific {@link #getOperationName} implementation.
+ * When using this Filter with Jersey, the full @Path template hierarchy is used as the
+ * operation name.
+ */
+@Slf4j
+@Provider
+public class JerseyServerFilter extends ServerFilter {
+
+  private Map<CacheKey, String> methodToPathCache = new ConcurrentHashMap<>();
+  private Field normalizedTemplate;
+
+  public JerseyServerFilter(Tracer tracer,
+                            TraceContext traceContext) {
+    super(tracer, traceContext);
+
+    try {
+      normalizedTemplate = UriTemplate.class.getDeclaredField("normalizedTemplate");
+      normalizedTemplate.setAccessible(true);
+    } catch (NoSuchFieldException e) {
+      //TODO: Add metrics for failure counts
+      log.error("Cannot access the normalizedTemplate field from UriTemplate. "
+                    + "Path operation name for tracing will be disabled.", e);
+    }
+  }
+
+  /**
+   * Retrieves an operation name that contains the http method and value of the
+   * {@link Path} used by a Jersey server implementation.
+   * @return the operation name
+   */
+  @Override
+  protected String getOperationName(ContainerRequestContext containerRequestContext) {
+    CacheKey key  = CacheKey.of(resourceInfo.getResourceMethod(), containerRequestContext.getMethod());
+    if(methodToPathCache.containsKey(key)){
+      return methodToPathCache.get(key);
+    }
+
+    try {
+      if (normalizedTemplate != null && containerRequestContext instanceof ContainerRequest) {
+        ContainerRequest containerRequest = (ContainerRequest) containerRequestContext;
+        ExtendedUriInfo uriInfo = containerRequest.getUriInfo();
+        if (uriInfo instanceof UriRoutingContext) {
+          UriRoutingContext uriRoutingContext = (UriRoutingContext) uriInfo;
+          List<UriTemplate> templates = uriRoutingContext.getMatchedTemplates();
+          String path = "";
+          for (UriTemplate uriTemplate : templates) {
+            String template = (String) normalizedTemplate.get(uriTemplate);
```

Comment:
can we instead look up the annotation via reflection (first on method, then on class) and read the route from it? It would be cleaner than depending on a private implementation detail.